core/arch/x86/decode.c

```diff
@@ -1223,14 +1347,21 @@ decode_reg(decode_reg_t which_reg, decode_info_t *di, byte optype, opnd_size_t o
     case TYPE_W:
     case TYPE_V_MODRM:
     case TYPE_VSIB:
-        return ((TEST(PREFIX_VEX_L, di->prefixes) &&
-                 /* Not only do we use this for .LIG (where raw reg is either
-                  * OPSZ_32 or OPSZ_16_vex32) but also for VSIB which currently
-                  * does not get up to OPSZ_16 so we can use this negative check.
-                  */
-                 expand_subreg_size(opsize) != OPSZ_16)
-                    ? (extend ? (REG_START_YMM + 8 + reg) : (REG_START_YMM + reg))
-                    : (extend ? (REG_START_XMM + 8 + reg) : (REG_START_XMM + reg)));
+      return (
+          TEST(PREFIX_EVEX_RR, di->prefixes)
+              ? (extend ? (DR_REG_START_ZMM + 16 + reg)
```

Comment:
Missing simd > 24. Let's don't mix the meaning of 'extend' or 'extended' being > 8 registers (rex.R) and the extension of AVX-512 from 16 to 32 simd registers anywhere in core/.